From 46e2b38f17535fdc31cfdc8bdc994dd8ca9d8722 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 26 Mar 2025 14:03:32 +0000
Subject: [PATCH] shell-search-provider: Don't build result when invalid ID
 specified

This fixes an error when invalid IDs are passed to the GetResultMetas D-Bus interface.
---
 src/bijiben-shell-search-provider.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/bijiben-shell-search-provider.c b/src/bijiben-shell-search-provider.c
index bcf5c07c..c9939c45 100644
--- a/src/bijiben-shell-search-provider.c
+++ b/src/bijiben-shell-search-provider.c
@@ -260,10 +260,14 @@ add_single_note_meta (BijibenShellSearchProviderApp *self,
 
   if (tracker_sparql_cursor_next (cursor, NULL, NULL))
   {
-    g_variant_builder_open (results, G_VARIANT_TYPE ("a{sv}"));
-
     /* NIE:URL (id) */
     url = tracker_sparql_cursor_get_string (cursor, 0, 0);
+    if (url == NULL)
+    {
+      g_object_unref (cursor);
+      return;
+    }
+    g_variant_builder_open (results, G_VARIANT_TYPE ("a{sv}"));
     g_variant_builder_add (results, "{sv}", "id", g_variant_new_string (url));
 
     /* NIE:TITLE (name) is the title pushed by libbiji */
-- 
GitLab

